client.go

```diff
@@ -680,6 +682,32 @@ func (c *Client) Mkdir(path string) error {
 	}
 }
 
+// MkdirAll creates a directory named path, along with any necessary parents,
+// and returns nil, or else returns an error.
+// If path is already a directory, MkdirAll does nothing and returns nil.
+// If path contains a regular file, an error is returned
+func (c *Client) MkdirAll(path string) error {
+	parts := ""
+	for _, p := range strings.Split(path, "/") {
+		if p == "" {
+			continue
+		}
+		parts += "/" + p
```

Comment:
String building with concatenation is O(n²) time and space. It should be avoided in tight loops.